tasks/prepare.js

```diff
@@ -33,23 +33,48 @@ const prepare = async (done) => {
     files.forEach(file => components[dir].push(file));
   });
 
-  return gulp.src('./templates/index.html')
-    .pipe($.replace('[/* SOURCES */]', JSON.stringify(components)))
-    .pipe($.replace('{/* DATA */}', JSON.stringify(JSON.parse(data))))
-    .pipe($.replace('{/* COLORS */}', JSON.stringify(JSON.parse(colors))))
+  $.util.log('Using template', $.util.colors.magenta(config.custom_template || config.template));
+
+  return gulp.src(config.custom_template || config.template, { cwd: config.custom_template ? config.project : '' })
```

Comment:
Why using two variables `config.custom_template` and `config.template` ?
What about checking if a pre-defined custom template exist and if yes, override `config.template` ?

We can also create a dedicated config for the `Current Working Directory` instead of deducting that if there is a custom template it is inside the project dir...

Reply:
I need this `cwd` parameter for Gulp 🤔